src/playground/utils/codemirror-linter-extension.js

```diff
@@ -271,14 +271,29 @@ function assignKeys(actions) {
 
 function renderDiagnostic(view, diagnostic) {
     const element = document.createElement("div");
+    const root = ReactDOM.createRoot(element);
 
-    ReactDOM.render(
-        <Popup
-            message={diagnostic.message}
-            onFix={diagnostic.actions && (() => diagnostic.actions[0].apply(view, diagnostic.from, diagnostic.to))}
-            ruleName={diagnostic.source.replace("jshint:", "")}
-        />,
-        element
+    // clean up ruleName first
+    // see https://github.com/codemirror/lang-javascript/blob/main/src/eslint.ts#L51
```

Comment:
This code can change on the `main` branch in the future, so let's use the permalink instead.

```suggestion
    // see https://github.com/codemirror/lang-javascript/blob/749ed7d353caab74996f3ad98c9c963a0ac646a7/src/eslint.ts#L51
```

Reply:
Makes sense.